src/components/SettingsPane/index.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import {
+	Box,
+	Card,
+	CardContent,
+	CardHeader,
+	Stack,
+	Typography,
+} from '@mui/material';
+import { color } from '@balena/design-tokens';
+
+export interface SettingsPaneProps {
+	title?: string | React.ReactNode;
+	children: React.ReactNode;
+	danger?: boolean;
+}
+
+export interface SettingsPaneItemProps {
+	title: string | React.ReactNode;
+	description?: string | React.ReactNode;
+	action?: React.ReactNode;
+}
+
+export const SettingsPane = ({
+	title,
+	children,
+	danger,
+}: SettingsPaneProps) => (
+	<Card>
+		{title && (
+			<CardHeader
+				title={title}
+				sx={{
+					'.MuiCardHeader-title': {
+						color: danger ? color.text.danger.value : 'inherit',
+					},
+				}}
+			/>
+		)}
+		<CardContent sx={{ '&:last-child': { pb: 0 } }}>{children}</CardContent>
+	</Card>
+);
+
+export const SettingsPaneItem = ({
+	title,
+	description,
+	action,
+}: SettingsPaneItemProps) => (
+	<Stack
+		gap={2}
+		direction={['column', 'column', 'row']}
+		justifyContent="space-between"
+		alignItems={['start', 'start', 'center']}
+		sx={{
+			'&:not(&:last-child)': {
+				borderBottom: `dashed ${color.border.subtle.value} 1px`,
+				pb: 3,
+				mb: 3,
+			},
+		}}
+	>
+		<Stack gap={2} maxWidth={[undefined, undefined, '70%']}>
+			<Typography variant="bodyLg">{title}</Typography>
+			<Typography variant="body" color="text.secondary">
+				{description}
+			</Typography>
+		</Stack>
+		<Box>{action}</Box>
```

Comment:
Why wrap it in a Box?

Reply:
To allow for a 70% / 30% split while keep the right button size (otherwise the button will take all 30%)